### PR TITLE
Integer

### PR DESCRIPTION
**Pull request by dupender**
Please review my code: Even, Integers, Fibonacci and corresponding unit tests.
Thank-you!

### CODE REVIEW

**Comment by scottpschaffer** [comment]
:+1:

**Comment by scottpschaffer** [comment]
@jcockhren Our code is ready for you to review.

**Comment by dupender** [comment]
@jcockhren Please review my pull request. Thank-you!

File: Mathmagician.Tests/EvenTest.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            //Assert()
+            Assert.AreEqual(8, my_even.EvenList.Count);   
+        }
+
+        [TestMethod]
+        public void TestEvenType()
+        {
+            Even my_even = new Even();
+            Assert.IsInstanceOfType(my_even, typeof(Even));
+        }
```
Comment (jcockhren) [inline Mathmagician.Tests/EvenTest.cs]: @dupender This test isn't necessary.

File: Mathmagician.Tests/FibonacciTest.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        [TestMethod]
+        public void FibNotNullTest()
+        {
+            Fibonacci my_fib = new Fibonacci();
+            my_fib.GenerateFibonacci(5);
+            int temp = my_fib.FibonacciList[0];
+
+            //Assert
+            Assert.IsNotNull(temp);
```
Comment (jcockhren) [inline Mathmagician.Tests/FibonacciTest.cs]: @dupender This test isn't useful. Maybe you should check for the first Fibonacci?

File: Mathmagician.Tests/FibonacciTest.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            Fibonacci my_fib = new Fibonacci();
+            my_fib.GenerateFibonacci(5);
+            int temp = my_fib.FibonacciList[3];
+
+            //Assert
+            Assert.IsTrue(temp == 2);
+        }
+
+        [TestMethod]
+        public void FibListTest1()
```
Comment (jcockhren) [inline Mathmagician.Tests/FibonacciTest.cs]: @dupender Come up with a better test name that describes this test.

File: Mathmagician.Tests/FibonacciTest.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            Fibonacci my_fib = new Fibonacci();
+            my_fib.GenerateFibonacci(5);
+
+            //Assert
+            CollectionAssert.AreEqual(new List <int>{0,1,1,2,3}, my_fib.FibonacciList);
+        }
+
+        [TestMethod]
+        public void FibListTest2()
```
Comment (jcockhren) [inline Mathmagician.Tests/FibonacciTest.cs]: @dupender Come up with a better test name that describes this test as well.

File: Mathmagician/Fibonacci.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            int a = 0, b = 1, c = 0;
+            Console.Write("{0} {1}", a, b);
+
+            for (int i = 2; i < numFib; i++)
+            {
+
+                c = a + b;
+                FibonacciList.Add(c);
+                Console.Write(" {0}", c);
```
Comment (jcockhren) [inline Mathmagician/Fibonacci.cs]: @dupender Your number class should not call the `Console` class. Remove this.

File: Mathmagician/Integer.cs [hunk 6]
```diff
@@ -8,5 +8,19 @@ namespace Mathmagician
...
+       // public int IntegerValue {get; set;}
+
+       public List<int> IntegerList = new List<int>();
+
+        public void GenerateInteger(int numInt)
+        {
+            for(int i = 0; i < numInt; i++)
+            {
+                IntegerList.Add(i);
+                Console.WriteLine(i);
```
Comment (jcockhren) [inline Mathmagician/Integer.cs]: @dupender remove this line.

**Comment by dupender** [comment]
@jcockhren Please review my incorporated changes

File: Mathmagician/Even.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            if (my_even % 2 == 0)
+            {
+                this.EvenValue = my_even;
+               
+            } else
+            {
+                throw new ArgumentException();
+            }
+        }*/
```
Comment (BlaiseGratton) [inline Mathmagician/Even.cs]: Be sure to remove all commented out old code

File: Mathmagician/Fibonacci.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+             //   Console.Write(" {0}", c);
+                a = b;
+                b = c;
+                
+            }
+            return FibonacciList;
+           /* for (int i=0; i<FibonacciList.Count; i++)
+            {
+                Console.WriteLine(FibonacciList[i]);
+            }*/
```
Comment (BlaiseGratton) [inline Mathmagician/Fibonacci.cs]: Remove comments

File: Mathmagician/Integer.cs [hunk 9]
```diff
@@ -6,7 +6,20 @@
...
     {
+       
+       private List<int> IntegerList = new List<int>();
+
+        public List<int> GenerateInteger(int numInt)
+        {
+            for(int i = 0; i < numInt; i++)
+            {
+                IntegerList.Add(i);
+               // Console.WriteLine(i);
```
Comment (BlaiseGratton) [inline Mathmagician/Integer.cs]: 🎱

File: Mathmagician.Tests/FibonacciTest.cs [hunk 10]
```diff
@@ -27,42 +27,40 @@ public void FibEnsureICanCreateAnInstance()
         public void FibNotNullTest()
```
Comment (jcockhren) [inline Mathmagician.Tests/FibonacciTest.cs]: @dupender this test is much better now, but needs to be renamed.

**Comment by jcockhren** [comment]
:+1:

**Pull request merged**